ion-schema/src/ion_path.rs

```diff
@@ -0,0 +1,90 @@
+use ion_rs::value::owned::Element;
+use ion_rs::value::Builder;
+use ion_rs::Symbol;
+use std::fmt;
+use std::fmt::Formatter;
+
+#[derive(Clone, PartialEq)]
+pub enum IonPathElement {
+    IndexedElement { index: usize },
+    Field { name: String },
```

Comment:
What's the rationale for using
```rust
pub enum IonPathElement {
  Index { index: usize },
  Field { name: String },
}
```
over
```rust
pub enum IonPathElement {
  Index(usize),
  Field(String),
}
```
?

Reply:
Usually with 2nd option, I would add a document comment explaining what the parameterized enum variant means but 1st one is self explanatory so I chose the second one. But it ended up having just one parameter(index or field name) so there's not really any benefit from using 1st option now.